[Jet Tan Ee Kiat] iP

medajet opened the pull request:
### JetBot

> “Object-oriented programming is an exceptionally bad idea which could only have originated in California.” – Edsger Wybe Dijkstra

JetBot frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~fast~~ SUPER fast to use

All you need to do is,

1. download it from [here](https://github.com/medajet/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Best of all, it's **free**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders

Here's a snippet of the code!
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Duke.class, args);
    }
```

Test it out by typing the following command in the following format:

`todo <task description>`

Thanks for using JetBot!

> **“Object-oriented programming is an exceptionally bad idea which could only have originated in California.”** – Edsger Wybe Dijkstra

Commit:
Add Gradle support

medajet committed:
Level-1

medajet committed:
Implement greet, echo and exit for Level-1

medajet committed:
Merge remote-tracking branch 'origin/master'

medajet committed:
Implement add and list for Level-2

medajet committed:
Implement Task class and ability to mark/unmark as done for Level-3 and A-Classes

medajet committed:
Implement ToDos, Events, Deadlines for Level-4

medajet committed:
Implement error handling for Level-5

medajet committed:
Implement delete function for Level-6, improve error handling

medajet committed:
Update testcases

medajet committed:
Change switch-case indentation

medajet committed:
Implement save/load feature for Level-7

medajet committed:
Add existence check for file/folder

medajet committed:
Add date/time support for Level-8

medajet committed:
Merge pull request #1 from medajet/branch-Level-7

Merge from branch-Level-7

medajet committed:
Merge pull request #2 from medajet/branch-Level-8

Merge from branch-Level-8

medajet committed:
Fix loader to account for date/time

medajet committed:
Refactor code to extract out closely related code as classes

medajet committed:
Divide classes into packages

medajet committed:
Add JUnit tests

medajet committed:
Implement find for Level-9

medajet committed:
Update javadocs

medajet committed:
Merge pull request #3 from medajet/branch-A-JavaDoc

Merge from branch-A-JavaDoc

medajet committed:
Merge pull request #4 from medajet/branch-Level-9

Merge from branch-Level-9

medajet committed:
Update indentation

medajet committed:
Merge pull request #5 from medajet/branch-A-CodingStandard

Merge from branch-A-CodingStandard

medajet committed:
Merge branch 'add-gradle-support' into A-Gradle

medajet committed:
Add gradle support for A-Gradle

medajet committed:
Implement GUI prototype for Level-10

medajet committed:
Merge pull request #6 from medajet/branch-A-Gradle

Merge from branch-A-Gradle

medajet committed:
Merge branch 'master' into branch-Level-10

medajet committed:
Merge pull request #7 from medajet/branch-Level-10

Merge from branch-Level-10

medajet committed:
Fix build.gradle

Harish-coding commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+package duke;
```
Deadline class looks great!, the file is properly packaged, and all the imports are explicitly mentioned! 👍

Harish-coding commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+    /**
+     * Returns a String containing the response based on user input.
+     *
+     * @param input the user input
+     * @return String containing the response based on user input
```
I think period is missing at the end of these line.....

Harish-coding commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,84 @@
+    }
+}
+
+class UnknownCommandException extends DukeException {
```
I believe that UnknownCommandException class should be defined in a separate file...... 🤔

Harish-coding commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,84 @@
+    }
+}
+
+class InvalidInputException extends DukeException {
```
I believe that InvalidInputException class should be defined in a separate file...... 🤔

Harish-coding commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,84 @@
+    }
+}
+
+class EmptyDescException extends DukeException {
```
I believe that EmptyDescException class should be defined in a separate file...... 🤔

Harish-coding commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,9 @@
+package duke;
+
+import javafx.application.Application;
+
```
Maybe you could add comments to include a description of what this class does....what do u think? 🤔

Harish-coding commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,52 @@
+package duke;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Button;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.image.Image;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.VBox;
```
Great job on explicitly mentioning all the classes imported! 👍

Harish-coding commented on src/main/java/duke/MainWindow.java:
```diff
@@ -0,0 +1,52 @@
+     * the dialog container. Clears the user input after processing.
+     */
+    @FXML
+    private void handleUserInput() {
```
good choice for method name!

Harish-coding commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,145 @@
+        String[] arr = input.split(" ");
+        String command = arr[0]; // first word of the user input
+        switch (command) {
+        case "bye":
```
Switch statement indentation is correctly followed according to module guidelines!! 😄

Harish-coding commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,145 @@
+            }
+            break;
+        case "mark": {
+            int num = Integer.parseInt(arr[1]);
```
just a suggestion maybe each case of the switch statements can be executed by a private function (each for a command) defined in parser class.

For example, for **case "mark":** , a function **_private void markTask(int num)_**, could handle the code execution. it may make it easier to read and also edit in the future.... what do u think ? 🤔

Harish-coding commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,145 @@
+            System.out.println(Ui.getBorder() + "Here are the matching tasks in your list:\n" + listString + Ui.getBorder());
+            break;
+        }
+        default:
```
great job at handling the user commands! 😄

Harish-coding commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,101 @@
+            if (isSuccess) {
+                System.out.println("Tasklist file has been created successfully.");
+            } else {
+                throw new IOException("Failed to create tasklist file, please try again.");
```
good job at anticipating and resolving IO exception error. 👍

Harish-coding reviewed (commented):
LGTM! The code is laid out very logically, and the readability is good! just a few minor suggestions to consider! 😄

medajet commented:
> LGTM! The code is laid out very logically, and the readability is good! just a few minor suggestions to consider! 😄

Thanks for the review comments Harish! All very valid suggestions, will take them into account 👍

zawedcvg commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,84 @@
+package duke;
+
+public class DukeException extends RuntimeException {
```
I believe this include some comments about what this file is about

zawedcvg commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+     * Constructor for an instance of the Chatbot.
+     */
+    public Duke() {
+        try {
```
maybe have one try-catch statement with both of the errors caught in one catch statement?

zawedcvg commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,57 @@
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            PrintStream printStream = new PrintStream(outputStream);
+
+            PrintStream oldStream = System.out;
```
Better variable name could help someone reading the code in the future.

zawedcvg commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The Event class.
```
This could include a bit more info about what the Event class is about. Otherwise this file looks great.

zawedcvg commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,101 @@
+ * @author Jet Tan
+ */
+public class Storage {
+    public static final String DATA_FOLDER_PATH = "./data";
```
Great variable names following the coding standards.

zawedcvg commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,57 @@
+     * @param tasknum the specified task to mark
+     */
+    public static void mark(int tasknum) {
+        Task t = taskList.get(tasknum - 1);
```
You could have a better name for the variable to increase readability

zawedcvg reviewed (commented):
Overall your code looks great and follows coding standards in most of the files.

medajet committed:
Update build.gradle for shadowJar command

Using the shadowJar command requires changes to the provided build.gradle file. Thus, necessary changes have been made to the file, such as correcting the mainClassName.

medajet committed:
Use assert in code for A-Assertions

medajet commented:
> Overall your code looks great and follows coding standards in most of the files.

Thanks for the comments! Will take the suggestions into account.

medajet committed:
Improve code quality for A-CodeQuality

After the PR review, several problems related to code quality have been identified and subsequently rectified in this commit.

medajet committed:
Merge pull request #8 from medajet/A-Assertions

Use assert in code for A-Assertions

medajet committed:
Merge pull request #9 from medajet/A-CodeQuality

Improve code quality for A-CodeQuality

medajet committed:
Implement task duplicate detection for BCD-Extension

medajet committed:
Merge pull request #10 from medajet/branch-BCD-Extension

Implement task duplicate detection for BCD-Extension

medajet committed:
Remove line border

medajet committed:
Add title to window

medajet committed:
Shorten code

medajet committed:
Add UI screenshot

medajet committed:
Fix build.gradle

medajet committed:
Update README.md

medajet committed:
Update README.md

medajet committed:
Update README.md

medajet committed:
Shorten code and fix bugs

medajet committed:
Shorten code

medajet committed:
Shorten code

medajet committed:
Shorten code

medajet committed:
Merge pull request #12 from medajet/branch-Refine

Branch refine

medajet committed:
Update javadoc

medajet committed:
Merge branch 'master' of https://github.com/medajet/ip into branch-Refine

medajet committed:
Add testcases
